synapseclient/models/mixins/table_operator.py

```diff
@@ -2428,63 +2699,250 @@ async def main():
+                prefix="chunked_csv_for_synapse_store_rows",
+                suffix=".csv",
+                dir=temp_dir,
+            ) as temp_file:
+                try:
+                    # TODO: This portion of the code should be updated to support uploading a file from memory using BytesIO (Ticket to be created)
+                    # TODO: The way that the dataframe is split up is also needed. As of right now we are writing the CSV, and then letting the CSV
+                    # TODO: upload process take over. However, if we are uploading from memory, then we don't have the ability to write to a file, and then
+                    # TODO: upload that file. We need to be able to split the dataframe into chunks, and then upload those chunks to Synapse.
+                    values.to_csv(
+                        temp_file.name,
+                        index=False,
+                        float_format="%.12g",
+                        **(to_csv_kwargs or {}),
+                    )
+                    # NOTE: reason for flat_format='%.12g':
+                    # pandas automatically converts int columns into float64 columns when some cells in the column have no
+                    # value. If we write the whole number back as a decimal (e.g. '3.0'), Synapse complains that we are writing
+                    # a float into a INTEGER(synapse table type) column. Using the 'g' will strip off '.0' from whole number
+                    # values. pandas by default (with no float_format parameter) seems to keep 12 values after decimal, so we
+                    # use '%.12g'.c
+                    # see SYNPY-267.
+
+                    with logging_redirect_tqdm(loggers=[client.logger]):
+                        await self._chunk_and_upload_csv(
+                            path_to_csv=temp_file.name,
+                            insert_size_byte=insert_size_byte,
+                            csv_table_descriptor=csv_table_descriptor,
+                            schema_change_request=schema_change_request,
+                            client=client,
+                            additional_changes=additional_changes,
+                            job_timeout=job_timeout,
+                        )
+                finally:
+                    temp_file.close()
+
+        else:
+            raise ValueError(
+                "Don't know how to make tables from values of type %s." % type(values)
             )
+
+    async def _chunk_and_upload_csv(
+        self,
+        path_to_csv: str,
+        insert_size_byte: int,
+        csv_table_descriptor: CsvTableDescriptor,
+        schema_change_request: TableSchemaChangeRequest,
+        client: Synapse,
+        job_timeout: int,
+        additional_changes: List[
+            Union[
+                "TableSchemaChangeRequest",
+                "UploadToTableRequest",
+                "AppendableRowSetRequest",
+            ]
+        ] = None,
+    ) -> None:
```

Comment:
I'm not a super big fan of the code I wrote for this. I had looked at some alternative libraries but was hesitant to add another dependency here, especially when thinking about performance concerns that are outlined in [this comparison for splitting CSV files](https://www.mungingdata.com/python/split-csv-write-chunk-pandas/#dask).

From their investigation, the Python filesystem API was much quicker (Likely since under the hood it uses C). I didn't do a new analysis based on the code I wrote here as I've chalked it up in a "Good enough" category.